src/electron/index.ts

```diff
@@ -194,24 +194,38 @@ function interceptShadowsocksLink(argv: string[]) {
 
 // Set the app to launch at startup to connect automatically in case of a shutdown while
 // proxying.
-async function enableAutoLaunch(): Promise<void> {
-  if (os.platform() === 'linux') {
-    if (process.env.APPIMAGE) {
+async function setupAutoLaunch(args: SerializableTunnel): Promise<void> {
+  try {
+    await tunnelStore.save(args);
+    if (os.platform() === 'linux') {
+      if (process.env.APPIMAGE) {
+        const outlineAutoLauncher = new autoLaunch({
+          name: 'OutlineClient',
+          path: process.env.APPIMAGE,
+        });
+        outlineAutoLauncher.enable();
+      }
+    } else {
+      app.setLoginItemSettings({openAtLogin: true, args: [Options.AUTOSTART]});
+    }
+  } catch (e) {
+    console.error(`Failed to set up auto-launch: ${e.message}`);
+  }
+}
+
+async function tearDownAutoLaunch() {
+  try {
+    if (os.platform() === 'linux') {
       const outlineAutoLauncher = new autoLaunch({
         name: 'OutlineClient',
-        path: process.env.APPIMAGE,
```

Comment:
Do you not need the path to disable?

Reply:
No, the auto launch config goes in a file based on the name only. The path is for the binary to be called.